kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -78,6 +80,24 @@ object Channel:
                 }
             }
 
+        /** Takes [[n]] elements from the channel, asynchronously blocking if necessary.
+          *
+          * @return
+          *   The taken element
+          */
+        def takeExactly(n: Int)(using Frame): Seq[A] < (Abort[Closed] & Async) =
+            Loop[Seq[A], Seq[A], Abort[Closed] & Async](Seq.empty): lastChunk =>
```

Comment:
Is there a performance overhead for Loop I should be concerned about?

Reply:
Yes. In fact, would it be alright with you to make all of these methods (`Queue#drain`, `Channel#drain`, `Channel#drainUpTo`, etc) explicitly yield `Chunk`? It seems appropriate for relatively low level APIs.

Reply:
Yep! I think we should specialize to Chunk where possible